http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/http2/Http2MultiplexedChannelPool.java

```diff
@@ -150,7 +174,27 @@ private void releaseParentChannel(Channel parentChannel) {
 
     @Override
     public void close() {
-        doInEventLoop(eventLoop, connectionPool::close);
+        try {
+            setClosedFlag().await();
+            for (MultiplexedChannelRecord c : connections) {
+                Future<Channel> f = c.getConnectionFuture();
+                f.await();
+                if (f.isSuccess()) {
+                    connectionPool.release(f.getNow()).await();
+                }
+            }
+            connectionPool.close();
+        } catch (InterruptedException ie) {
+            throw new RuntimeException(ie);
+        }
     }
 
+    private Promise<Void> setClosedFlag() {
```

Comment:
Is there an advantage to doing it this way versus using an AtomicBoolean or something? Just to let the acquires submitted before close called to get a chance to acquire?

Reply:
Yeah, I wanted to make sure acquire and close serialized so it's not possible to close while in the middle of an acquire.

Reply:
+1, that makes sense.